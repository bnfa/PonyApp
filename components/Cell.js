import React from 'react';
import {View, StyleSheet, Text} from 'react-native';

/**
 * Cell Component
 * @param {array} borders Array of string containing ["norht", "west"]
 * @param {boolean} isDomokun true if the Cell index matches the Domokun position
 * @param {boolean} isEndPoint true if the Cell index matches the End-Point position
 * @param {boolean} isPony true if the Cell index matches the Pony position
 */

function Cell({borders, isDomokun, isEndPoint, isPony}) {
  let cellStyle = [styles.cell];
  for (const value of borders) {
    cellStyle.push(styles[value]);
  }
  return (
    <View style={cellStyle}>
      {isDomokun && <Text>D</Text>}
      {isEndPoint && <Text>E</Text>}
      {isPony && <Text>P</Text>}
    </View>
  );
}

const styles = StyleSheet.create({
  cell: {
    width: 50,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  north: {
    borderTopColor: 'black',
    borderTopWidth: 1,
  },
  west: {
    borderLeftColor: 'black',
    borderLeftWidth: 1,
  },
});
export default Cell;
