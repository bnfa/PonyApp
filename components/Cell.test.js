import React from 'react';
import renderer from 'react-test-renderer';
import Cell from './Cell';

test('Cell renders correctly', () => {
  const tree = renderer
    .create(
      <Cell
        borders={['north']}
        isDomokun={true}
        isEndPoint={false}
        isPony={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
