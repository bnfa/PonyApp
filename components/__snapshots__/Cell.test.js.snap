// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      },
      Object {
        "borderTopColor": "black",
        "borderTopWidth": 1,
      },
    ]
  }
>
  <Text>
    D
  </Text>
</View>
`;
